packages/core/src/generators/html.ts

```diff
@@ -723,30 +726,37 @@ export const componentToCustomElement =
           }
         }
 
+        onUpdate() {
+          ${
+            !json.hooks.onUpdate?.length
+              ? ''
+              : `
+      ${json.hooks.onUpdate.map((hook) =>
+        updateReferencesInCode(hook.code, useOptions),
+      )} 
+      `
+          } 
+        }
+
         update() {
-          ${Object.keys(useOptions.onChangeJsById)
-            .map((key) => {
-              const value = useOptions.onChangeJsById[key];
-              if (!value) {
-                return '';
-              }
-              return `
+          this.onUpdate();
+          this.updateBindings();
+        }
+
+        updateBindings {
```

Comment:
I think you want this to be a function:

```suggestion
        updateBindings() {
```